x/evm/keeper/params.go

```diff
@@ -49,7 +50,44 @@ func (k Keeper) SetParams(ctx sdk.Context, params types.Params) error {
 
 // GetLegacyParams returns param set for version before migrate
 func (k Keeper) GetLegacyParams(ctx sdk.Context) types.Params {
-	var params types.Params
-	k.ss.GetParamSetIfExists(ctx, &params)
+	var legacyParams legacytypes.LegacyParams
+	k.ss.GetParamSetIfExists(ctx, &legacyParams)
```

Comment:
I might need a `nil` check here to prevent a panic further down if there are no params set

Reply:
Will add a test case for that to see